cider/tests/functionality/CiderHashJoinTest.cpp

```diff
@@ -272,24 +272,64 @@ HASH_JOIN_TEST_UNIT(CiderOneToOneSeqJoinTest, ExprJoinSeqTest2, *, d, = 2 *)
 HASH_JOIN_TEST_UNIT(CiderOneToOneSeqJoinTest, ExprJoinSeqTest3, *, d, *2 = 2 *)
 
 // using OR to avoid 0 results
-#define DOUBLE_JOIN_CONDITION_TEST_UNIT(TEST_CLASS, UNIT_NAME, PROJECT)              \
+// TODO: (spevenhe) OR will fail back to loop join while AND is still hash join
+#define DOUBLE_JOIN_OR_CONDITION_TEST(TEST_CLASS, UNIT_NAME, PROJECT)               \
+  TEST_F(TEST_CLASS, UNIT_NAME) {                                                   \
+    /*INNER JOIN ON INTEGER OR FLOAT*/                                              \
+    assertJoinQueryRowEqualAndReset(                                                \
+        "SELECT " #PROJECT                                                          \
+        " from table_probe JOIN table_hash ON l_b = r_b OR l_c =  r_c ");           \
+    /*LEFT JOIN ON BIGINT OR DOUBLE*/                                               \
+    assertJoinQueryRowEqualAndReset(                                                \
+        "SELECT " #PROJECT                                                          \
+        " from table_probe LEFT JOIN table_hash ON l_a = r_a OR l_d = r_d ");       \
+    /*INNER JOIN ON INTEGER OR CONSTANT*/                                           \
+    assertJoinQueryRowEqualAndReset(                                                \
+        "SELECT " #PROJECT                                                          \
+        " from table_probe JOIN table_hash ON l_b = r_b OR l_b = 10 ");             \
+    /*LEFT JOIN ON BIGINT OR CONSTANT*/                                             \
+    assertJoinQueryRowEqualAndReset(                                                \
+        "SELECT " #PROJECT                                                          \
+        " from table_probe LEFT JOIN table_hash ON l_a = r_a OR l_a = 10 ");        \
+    /*INNER JOIN ON INTEGER AND NOT NULL*/                                          \
+    assertJoinQueryRowEqualAndReset(                                                \
+        "SELECT " #PROJECT                                                          \
+        " from table_probe JOIN table_hash ON l_b = r_b OR l_b IS NOT NULL ");      \
+    /*LEFT JOIN ON BIGINT AND NOT NULL*/                                            \
+    assertJoinQueryRowEqualAndReset(                                                \
+        "SELECT " #PROJECT                                                          \
+        " from table_probe LEFT JOIN table_hash ON l_a = r_a OR l_a IS NOT NULL "); \
+  }
+
+// DOUBLE_JOIN_OR_CONDITION_TEST(CiderOneToOneSeqJoinTest, ORJoinConditionTest1, *)
```

Comment:
why disable these?

Reply:
Unlike ```AND``` which uses hash join, conditions with ```OR``` will fall back to loop join. Haven't investigated the reason yet

Reply:
Please add some comments why we comment this out.